### ISSUE TITLE
[ARM CI]Hit the aarch64 

### ISSUE THREAD

**Issue by bzhaoopenstack** [greenplum-db/gpdb#10962]
### Greenplum version or build
master branch.
On aarch64

I test on aarch64 and X86 platform with the same OS version, only aarch64 hits this.
Can trace this from https://logs.openlabtesting.org/logs/periodic-12/github.com/greenplum-db/gpdb/master/gpdb-installcheck-world-tests-on-arm64/f49aba0/test_results/

### OS version and uname -a
Ubuntu 1804

### Installation information ( pg_config )
```
./configure --with-openssl --with-ldap --with-libcurl --prefix="/home/zuul/gpdb/gpsql" --disable-orca --disable-gpcloud --disable-pxf --without-readline --with-python --with-gssapi  --enable-orafce --enable-mapreduce --with-perl
```

### Expected behavior
All regression tests PASS.

### Actual behavior
6 regression tests Fail due to the `assumed` different platforms have different sensitivity to float/long type values.

### Step to reproduce the behavior
rebuild and run the installcheck-world on aarch64.
```
source gpsql/greenplum_path.sh
make -C gpAux/gpdemo cluster
source gpAux/gpdemo/gpdemo-env.sh
make installcheck-world
```

Relies on above tests, do you guys have any good ideas? As we @amitdkhan-pg have faced this in java JDK, but it's the first time for C project. Thanks.

**Comment by bzhaoopenstack** [comment]
And I trace this from 2 commits.
The PG12 support commit https://github.com/greenplum-db/gpdb/commit/19cd1cf4b68faff2e29bc2fa884c480e4644cdb4
And the previous commit https://github.com/greenplum-db/gpdb/blob/8ae22d15ebf0cf68b0854e2a2630b7b997e3ce1c/src/test/regress/expected/percentile.out towards the same issue in src/test/regress/expected/percentile.out

Looks like the old behavior is more human-readable. Not sure why change the expected output files.

**Comment by bzhaoopenstack** [comment]
Sorry to disturb you again. ;-(. But seems I really need help from you. @hlinnaka  Thank you

**Comment by asimrp** [comment]
Note: `percentile.sql` is Greenplum specific, so changing anything there is easier.

A smaller repro to highlight the problem:
```SQL
create table test as select a from generate_series(1,9)a;
```
Use 0.8th percentile but in descending order of input:
```SQL
select percentile_cont(0.8) within group (order by a desc) from test;
  percentile_cont
--------------------
 2.5999999999999996
(1 row)
```
Should it be the same as 0.2nd percentile with ascending order of input?
```SQL
select percentile_cont(0.2) within group (order by a) from test;
 percentile_cont
-----------------
             2.6
(1 row)
```

On 6X, the answer for both the queries is 2.6.

**Comment by asimrp** [comment]
And on PostgreSQL v12 the behavior is identical to that on master branch.

**Comment by amitdkhan-pg** [comment]
I think these diffferences in float values might be expected across architectures or machines for approximate types. One easy option is to elliminate the possibility of inconsistent values from being printed in the output file. There are some other regression tests where extra_float_digits  is used to deal with these problems. We can do the same thing for percentile.sql

**Comment by asimrp** [comment]
Is it better to use numeric type instead?
```SQL
select percentile_cont(0.2) within group (order by a) :: numeric,
        percentile_cont(0.8) within group (order by a desc) :: numeric from perct group by b order by b;
```

**Comment by bzhaoopenstack** [comment]
@asimrp , hi , I test on both X86 and aarch64 platform with the same version `PostgreSQL 12.4` . I re-run you said test sqls in the previous comment. They are the same from pgsql side on these 2 ARCH. It has the same behavior on the pgsql master branch with the 2 ARCH.
```
select percentile_cont(0.8) within group (order by a desc) from test;
select percentile_cont(0.2) within group (order by a desc) from test;
```
Hmm, so it looks not a pgsql issue.

**Comment by hlinnaka** [comment]
The output changed as result of upstream commit 02ddd499322ab6f2f0d58692955dc9633c2150fc.

What exactly is the problem here? Are you seeing a different result on aarch64 than on x86-64? What is the result on each architeture?

**Comment by bzhaoopenstack** [comment]
https://logs.openlabtesting.org/logs/periodic-12/github.com/greenplum-db/gpdb/master/gpdb-installcheck-world-tests-on-arm64/f49aba0/test_results/regress_regression.diffs
Looks all `+` is the aarch64 result, `-` is the X86 result for now, the ARM CI failed a long time due to the difference.

@hlinnaka, Thanks, yeah, @amitdkhan-pg also mentioned `extra_float_digits ` might solve this. I'm seeing how to do in the next.

**Comment by amitdkhan-pg** [comment]
> Is it better to use numeric type instead?
> 
> ```sql
> select percentile_cont(0.2) within group (order by a) :: numeric,
>         percentile_cont(0.8) within group (order by a desc) :: numeric from perct group by b order by b;
> ```

Yeah, that should also work, but by setting the extra_float_digits at an initial place, we can as well prevent any other queries or any future queries in the file from having to always cast such values.

**Comment by amitdkhan-pg** [comment]
PR is created for fixing this issue : https://github.com/greenplum-db/gpdb/pull/11224
